src/xmipp/libraries/dimred/matrix_dimred.cpp

```diff
@@ -102,7 +102,7 @@ void ProgDimRed::defineParams()
     addParamsLine("                  CorrDim: Correlation dimension");
     addParamsLine("                  MLE: Maximum Likelihood Estimate");
     addParamsLine("  [--saveMapping <fn=\"\">] : Save mapping if available (PCA, LLTSA, LPP, pPCA, NPE) so that it can be reused later (Y=X*M)");
-    addParamsLine("                            :+X is the input matrix with individuals as rows");
+    addParamsLine("                            :+X is the input matrix with individuals as rows (mean subtracted from the columns)");
```

Comment:
The the param is the residual after the mean has been subtracted, or, the param is the subtracted mean? Can you clarify the description?

Reply:
I simply wanted to give a hint. But it s a good idea to explain it. I will let you evaluate what is necessary to say from the following:
Say `X_original` is the input matrix that each of its rows is a data point (before reducing the dimensions).
`X` is obtained from `X_original` by subtracting the mean of each column (row-wise).
A python code to explain the how to get `X`:
```
from numpy import loadtxt
X_original = loadtxt('data.txt')
X = X_original - outer(ones(X_original.shape[0]),mean(X_original, axis=0))
```

Reply:
I see. So how about sth like:
X is the input matrix (one data point before reducing the dimensions per row) with the mean deducted, i.e. X = X_orig - u(x).

Other option

Reply:
Actually, the mean should be deducted collectively from the matrix columns. We calculate the mean of each column separately, then we subtract the mean of each column from its rows.
Say the original matrix is of dimensions (1000 rows, 5 columns), then we will have 5 column means, each to be reduced from the corresponding 1000 elements in its row.